Extension/src/common.ts

```diff
@@ -736,13 +738,7 @@ export function allowExecution(file: string): Promise<void> {
                         if (isExec) {
                             resolve();
                         } else {
-                            fs.chmod(file, '755', (err: NodeJS.ErrnoException | null) => {
-                                if (err) {
-                                    reject(err);
-                                    return;
-                                }
-                                resolve();
-                            });
+                            chmodAsync(file, '755');
```

Comment:
I think this whole function should be made async now that you have `chmodAsync` implemented.  You can await this here as well as `checkFileExists` and `isExecutable`.